packages/flutter/lib/src/cupertino/segmented_control.dart

```diff
@@ -148,11 +168,28 @@ class CupertinoSegmentedControl<T extends Object> extends StatefulWidget {
   /// Defaults to the selectedColor at 20% opacity if null.
   final Color? pressedColor;
 
+  /// The color used to fill the background of the segment when it is disabled.
+  ///
+  /// Defaults to [CupertinoTheme]'s `primaryContrastingColor` with 50% opacity if null.
```

Comment:
We don't have specs for this behavior, so I'm actually not super sure how we should set this default value. So probably we can leave this nullable, and if a selected segment is disabled, we can return the background color to `widget.disabledColor ?? _selectedColor.widthOpacity(0.5)`; for unselected segment, I don't think there's any difference between enabled or disabled based on my experiment, so probably we can just return `widget.disabledColor ?? _unselectedColor`:

https://github.com/user-attachments/assets/49b6582a-4324-4b44-a6a4-088a8b549ca4

Reply:
I'm a bit confused here. I think when a segment is disabled, it will not be selected anymore, so it should be a single color only (without depending on the selected/unselected state). What if we set `disabledColor` to 50% opacity of `unselectedColor` on `_updateColors()`?

```dart
final Color disabledColor = widget.disabledColor ?? unselectedColor.withOpacity(0.5);
```

Reply:
> I'm a bit confused here. I think when a segment is disabled, it will not be selected anymore,

Sorry that the demo above looks a bit confusing. It's used to see the specs on the right side, it's not really "selecting" any segment:) And you are right, when a segment is disabled, it should not be selected anymore, but if a segment is initially selected, and then we programmatically disable it, it's possible that a segment is disabled and selected at the same time, does this make sense?

Reply:
Can you please share the tool you use to inspect colors? I would love to use it for later case :).

For the both selected and unselected color when segment is disabled, you are right. I captured iOS native Segment Control in selected/unselected states for the first disabled segment here:

![first selected](https://github.com/user-attachments/assets/a684d6f8-3e7f-47e7-a6c8-5e05011b6171)
![first unselected](https://github.com/user-attachments/assets/41b4ebfc-9ba8-49bf-8202-b23ec13bbc00)
![second selected](https://github.com/user-attachments/assets/6cb5f847-6c5d-4f00-99ab-f47d6cacee27)

Reply:
> Can you please share the tool you use to inspect colors? I would love to use it for later case :).

Ah it's just Xcode! When we run the app, there's a view hierarchy button which can be used to see all the visual details:)
![Screenshot 2024-10-23 at 10 45 10 AM](https://github.com/user-attachments/assets/04fa0220-f286-4712-b148-221a94dfdcff)